### PR TITLE
Feature/sdl 0188 interior vehicle data resumption

### PR DESCRIPTION
**Pull request by KhrystynaDubovyk**
Guidelines updates according to the proposal [SDL-0188](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0188-get-interior-data-resumption.md) #[2661](https://github.com/smartdevicelink/sdl_core/issues/2661)

Please, note this PR is based on https://github.com/smartdevicelink/sdl_hmi_integration_guidelines/pull/224

This PR is **ready** for review.

### Summary
**Changelog**
BC.OnAppRegistered - update index.md
RC.GetInteriorVehicleData - update index.md

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: docs/BasicCommunication/OnAppRegistered/index.md [hunk 1]
```diff
@@ -34,6 +34,19 @@ If the application resumes data successfully:
...
+
+If HMI responds with any kind of error or does not respond to any requests sent during resumption, SDL must revert already restored data with appropriate RPCs:
+    * `DeleteCommand`
+    * `DeleteSubMenu`
+    * `DeleteInteractionChoiceSet`
+    * `ResetGlobalProperties`
+    * `UnsubscribeButton`
+    * `UnsubscribeVehicleData`
+    * `UnsubscribeWayPoints`
+    * `DeleteWindow`
```
Comment (yang1070) [inline docs/BasicCommunication/OnAppRegistered/index.md:49]: please add GetInteriorVehicleData(moduleType, moduleID, subscribe=false)

File: docs/BasicCommunication/OnAppRegistered/index.md [hunk 2]
```diff
@@ -34,6 +34,19 @@ If the application resumes data successfully:
     * `SetGlobalProperties`
     * `SubscribeButton`
     * `SubscribeVehicleData`
+    * `SubscribeWayPoints`
+    * `CreateWindow`
+    * `GetInteriorVehicleData`
+
+If HMI responds with any kind of error or does not respond to any requests sent during resumption, SDL must revert already restored data with appropriate RPCs:
```
Comment (yang1070) [inline docs/BasicCommunication/OnAppRegistered/index.md:41]: If HMI responds with any kind of error or does not respond to any requests, or responds with success result code but failed to subscribe to interior vehicle data for a RC module sent during resumption, SDL must revert already restored data with appropriate RPCs:

File: docs/Navigation/SubscribeWayPoints/assets/SubscribeWayPoints_Resumption.txt [hunk 3]
```diff
@@ -0,0 +1,22 @@
+participant App
```
Comment (yang1070) [inline docs/Navigation/SubscribeWayPoints/assets/SubscribeWayPoints_Resumption.txt:1]: this file as well as file SubscribeWayPoints_Resumption.png does not belong to SDL188

File: docs/Navigation/SubscribeWayPoints/index.md [hunk 4]
```diff
@@ -15,6 +15,10 @@ Purpose
 Send future waypoint updates to SDL using [Navigation.OnWayPointChange](../onwaypointchange)
 !!!
 
+!!! note
```
Comment (yang1070) [inline docs/Navigation/SubscribeWayPoints/index.md:18]: this does no belong to SDL188

File: docs/Navigation/SubscribeWayPoints/index.md [hunk 5]
```diff
@@ -32,6 +36,11 @@ SubscribeWayPoints
 ![SubscribeWayPoints](./assets/SubscribeWayPoints.png)
 |||
 
+|||
```
Comment (yang1070) [inline docs/Navigation/SubscribeWayPoints/index.md:39]: this does not belong to SDL188

File: docs/RC/GetInteriorVehicleData/index.md [hunk 6]
```diff
@@ -39,6 +41,13 @@ If the parameter `subscribe` is set to true, the mobile application has requeste
 HMI must provide optional `moduleId` param  in GetInteriorVehicleData_response.
 HMI must return in GetInteriorVehicleData_response the current value of the display mode used in HMI if `moduleType = HMI_SETTINGS` .
 
+!!! NOTE
+
+If during resumption HMI responds with error to GetInteriorVehicleData request SDL reverts already subscribed data and fails resumption, removing information about this subscription.
```
Comment (yang1070) [inline docs/RC/GetInteriorVehicleData/index.md:46]: If during resumption HMI responds with error to GetInteriorVehicleData request or responds with SUCCESS to GetInteriorVehicleData  but with parameter isSubscribed=false, SDL reverts already subscribed data and fails resumption for related application(s), removing information about this subscription.

File: docs/UI/AddCommand/assets/AddCommand_Resumption.txt [hunk 7]
```diff
@@ -0,0 +1,37 @@
+participant App
```
Comment (yang1070) [inline docs/UI/AddCommand/assets/AddCommand_Resumption.txt:1]: this does not belong to SDL188

File: docs/UI/AddCommand/index.md [hunk 8]
```diff
@@ -24,7 +24,7 @@ UI.AddCommand represents a request from an application to add a command to the a
 
   * If SDL sends the HMI a UI.AddCommand and a VR.AddCommand, and receives a SUCCESS from one and a failure from the other, SDL will send a UI.DeleteCommand for the AddCommand which succeeded.
   * If the `menuParams` contains a `parentID` the command is part of a sub menu. SDL adds SubMenu Commands to the top level Menu via [UI.AddSubMenu](../addsubmenu)
-  
+  * During data resumption SDL sends UI.AddCommands to HMI by `internal_consecutiveNumber` in the same order as they were created by mobile app in previous ignition cycle.
```
Comment (yang1070) [inline docs/UI/AddCommand/index.md:27]: this does not belong to SDL188

File: docs/UI/AddCommand/index.md [hunk 9]
```diff
@@ -91,6 +91,11 @@ AddCommand UI No Response, <abbr title="Voice Recognition">VR</abbr> Succeeds
 ![AddCommand](./assets/AddCommandUINoResponseVRSuccess.png)
 |||
 
+|||
+AddCommand restoring during data resumption
```
Comment (yang1070) [inline docs/UI/AddCommand/index.md:95]: this does not belong to SDL188

File: docs/VR/AddCommand/assets/AddCommandResumption.txt [hunk 10]
```diff
@@ -0,0 +1,37 @@
+participant App
```
Comment (yang1070) [inline docs/VR/AddCommand/assets/AddCommandResumption.txt:1]: this does not belong to SDL188

File: docs/VR/AddCommand/index.md [hunk 11]
```diff
@@ -27,6 +27,8 @@ The parameter `type` included in the `VR.AddCommand` request is the differentiat
 
 `Command` type <abbr title="Voice Recognition">VR</abbr> Commands should only be used when the user wants to interact with the App's Menu.
 
+During data resumption SDL sends VR.AddCommands to HMI by `internal_consecutiveNumber` in the same order as they were created by mobile app in previous ignition cycle.
```
Comment (yang1070) [inline docs/VR/AddCommand/index.md:30]: this does not belong to SDL188

File: docs/VR/AddCommand/index.md [hunk 12]
```diff
@@ -72,6 +74,11 @@ UI.AddCommand no response, VR.AddCommand returns SUCCESS
 ![AddCommand](./assets/AddCommandSuccessUINoResponse.png)
 |||
 
+|||
+AddCommand restoring during data resumption
```
Comment (yang1070) [inline docs/VR/AddCommand/index.md:78]: this does not belong to SDL188

File: docs/VehicleInfo/SubscribeVehicleData/index.md [hunk 13]
```diff
@@ -101,6 +103,17 @@ SubscribeVehicleData
 ![SubscribeVehicleData](./assets/SubscribeVehicleData.jpg)
 |||
 
+|||
```
Comment (yang1070) [inline docs/VehicleInfo/SubscribeVehicleData/index.md:106]: the diagrams are for SDL190, not for SDL188

File: docs/VehicleInfo/UnsubscribeVehicleData/index.md [hunk 14]
```diff
@@ -107,6 +109,11 @@ UnsubscribeVehicleData unexpected disconnect
 ![UnsubscribeVehicleData](./assets/UnsubscribeVehicleDataDisconnect.jpg)
 |||
 
+|||
```
Comment (yang1070) [inline docs/VehicleInfo/UnsubscribeVehicleData/index.md:112]: this is for SDL190, not for SDL188

File: docs/RC/GetInteriorVehicleData/index.md (hunk 6 above)
Comment (KhrystynaDubovyk) [reply docs/RC/GetInteriorVehicleData/index.md]: thank you, done in https://github.com/smartdevicelink/sdl_hmi_integration_guidelines/pull/225/commits/9074c7d61dea3f16ee213393369743b367f7b50d

File: docs/BasicCommunication/OnAppRegistered/index.md (hunk 1 above)
Comment (KhrystynaDubovyk) [reply docs/BasicCommunication/OnAppRegistered/index.md:49]: thank you, done in https://github.com/smartdevicelink/sdl_hmi_integration_guidelines/pull/225/commits/9074c7d61dea3f16ee213393369743b367f7b50d

**Comment by KhrystynaDubovyk** [comment]
@yang1070, please note, as I mentioned in https://github.com/smartdevicelink/sdl_hmi_integration_guidelines/pull/225#issue-469646545 this PR is based on #224, that's why you can see updates relevant for SDL-0190 feature.
Once the project maintainer merges #224, 
in this PR will be left only commits relevant for SDL-0188 feature - they are - https://github.com/smartdevicelink/sdl_hmi_integration_guidelines/pull/225/commits/e371363aed57bd2520ff7ef48e0088b9df9623b8 and  https://github.com/smartdevicelink/sdl_hmi_integration_guidelines/pull/225/commits/9074c7d61dea3f16ee213393369743b367f7b50d

File: docs/BasicCommunication/OnAppRegistered/index.md (hunk 2 above)
Comment (KhrystynaDubovyk) [reply docs/BasicCommunication/OnAppRegistered/index.md]: done in https://github.com/smartdevicelink/sdl_hmi_integration_guidelines/commit/9074c7d61dea3f16ee213393369743b367f7b50d

**Comment by KhrystynaDubovyk** [comment]
@yang1070 please, see the updates made

File: docs/RC/GetInteriorVehicleData/index.md [hunk 15]
```diff
@@ -19,6 +19,8 @@ Otherwise, SDL responds to the request with the cached data without forwarding i
 
 The HMI should return interior vehicle data that corresponds to the requested module (`moduleId` + `moduleType`).
 
+During data resumption process SDL sends GetInteriorVehicleData(subscribe=true) request to HMI and stores data received from HMI in cache.
```
Comment (ShobhitAd) [inline docs/RC/GetInteriorVehicleData/index.md:22]:
```suggestion
During the data resumption process, SDL sends GetInteriorVehicleData(subscribe=true) request to the HMI and stores data received from the HMI in cache.
```

File: docs/RC/GetInteriorVehicleData/index.md [hunk 16]
```diff
@@ -39,6 +41,13 @@ If the parameter `subscribe` is set to true, the mobile application has requeste
 HMI must provide optional `moduleId` param  in GetInteriorVehicleData_response.
 HMI must return in GetInteriorVehicleData_response the current value of the display mode used in HMI if `moduleType = HMI_SETTINGS` .
 
+!!! NOTE
+
+If during resumption HMI responds with error to GetInteriorVehicleData request or responds with SUCCESS to GetInteriorVehicleData but with parameter isSubscribed=false, SDL reverts already subscribed data and fails resumption for related application(s), removing information about this subscription. 
+In the case after reverting the subscription, there is no application subscribed to a certain module type, SDL sends GetInteriorVehicleData (subscribe=false) to HMI and clears cache for this module type.
```
Comment (ShobhitAd) [inline docs/RC/GetInteriorVehicleData/index.md:47]:
```suggestion
After reverting the subscription, if there is no application subscribed to a certain module type, SDL sends GetInteriorVehicleData (subscribe=false) to the HMI and clears the cache for this module type.
```

**Comment by theresalech** [comment]
@KhrystynaDubovyk, can you please confirm if this PR is ready for Livio's re-review?

**Comment by KhrystynaDubovyk** [comment]
@theresalech, yes, it is ready. thank you.

**Review by ShobhitAd — APPROVED** [review]

**Comment by KhrystynaDubovyk** [comment]
@ShobhitAd, the conflicts were resolved. The PR is ready for merging.

**Review by ShobhitAd — APPROVED** [review]

**Pull request merged**